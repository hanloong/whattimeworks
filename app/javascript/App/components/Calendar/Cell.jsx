import React from 'react'

export const Cell = ({ time }) => {
  return <td key={ time.zone }>
    { time.time }
  </td>
}
