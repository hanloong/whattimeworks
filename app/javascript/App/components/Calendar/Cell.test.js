import React from 'react';
import Cell from './Cell';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

test('Print out a table cell', () => {
  const time = { time: 'Today, 3pm', zone: 'Australia/Sydney' }
  const component = renderer.create(
    <MuiThemeProvider>
      <Cell time={ time } />
    </MuiThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
