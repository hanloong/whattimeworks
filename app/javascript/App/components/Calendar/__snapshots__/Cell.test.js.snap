// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Print out a table cell 1`] = `
<td
  className={undefined}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "fontSize": 13,
      "height": 48,
      "muiPrepared": true,
      "overflow": "hidden",
      "paddingLeft": 24,
      "paddingRight": 24,
      "textAlign": "left",
      "textOverflow": "ellipsis",
      "whiteSpace": "nowrap",
    }
  }
>
  Today, 3pm
</td>
`;
